docs/api/ethereum.md

```diff
@@ -1,5 +1,645 @@
+
+**Result:**
+
+A `boolean` which tells if the provider can be used.
+
+**Example:**
+
+```ts
+// perform query
+const isSupported = provider.isSupported();
+```
+
+### isSignedIn()
+
+A `synchronous` class instance `function` which returns `true` when user is signed into Bitski.
+
+**Result:**
+
+A `boolean` which tells if the user is signed into Bitski.
+
+**Example:**
+
+```ts
+// perform query
+const isSignedIn = provider.isSignedIn();
+```
+
+
+### isUnsafeRecipientId(ledgerId)
+
+A `synchronous` class instance `function` which returns `true` when the provided `id` is listed among unsafe recipient ids on the provided.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| ledgerId | [required] A `string` representing an address of the ERC-721 related smart contract on the Ethereum blockchain.
+
+**Result:**
+
+A `boolean` which tells if the `id` is unsafe recipient.
+
+**Example:**
+
+```ts
+// unsafe recipient address
+const criptoKittiesId = '0x06012c8cf97bead5deae237070f9587f8e7a266d';
+
+// perform query
+const isUnsafe = provider.isUnsafeRecipientId(criptoKittiesId);
+```
+
+**See also:**
+
+[unsafeRecipientIds](#unsaferecipientids)
+
+### mutationTimeout
+
+A class instance `variable` holding an `integer` number of milliseconds in which a mutation times out.
+
+### on(event, handler);
+
+A `synchronous` class instance `function` which attaches a new event handler.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| handler | [required] A callback `function` which is triggered on each `event`. When the `event` equals `ACCOUNT_CHANGE`, the first argument is a new account ID and the second is the old one, when the `event` equals `NETWORK_CHANGE`, the first argument is a new network version and the second is the old one.
+
+**Result:**
+
+An instance of the same provider class.
+
+**Example:**
+
+```ts
+import { ProviderEvent } from '@0xcert/ethereum-bitski-frontend-provider'; // or another provider
+
+provider.on(ProviderEvent.NETWORK_CHANGE, (networkVersion) => {
+  console.log('Network has changed', networkVersion);
+});
+```
+
+**See also:**
+
+[once (event, handler)](#once-event-handler), [off (event, handler)](#off-event-handler)
+
+### once(event, handler);
+
+A `synchronous` class instance `function` which attaches a new event handler. The event is automatically removed once the first `event` is emitted.
+
+**Arguments:**
+
+| Argument | Description
+|-|-
+| event | [required] A `string` representing a [provider event](./ethereum.md#provider-events) name.
+| handler | [required] A callback `function` which is triggered on each `event`. When the `event` equals `ACCOUNT_CHANGE`, the first argument is a new account ID and the second is the old one, when the `event` equals `NETWORK_CHANGE`, the first argument is a new network version and the second is the old one.
```

Comment:
`| handler | [required] A callback `function` which is triggered on each `event`. When the `event` equals `ACCOUNT_CHANGE`, the first argument is a new account ID and the second argument is the old ID, when the `event` equals `NETWORK_CHANGE`, the first argument is a new network version and the second argument is the old version.`